Classes/TMDebugConsole.m

```diff
@@ -185,31 +185,17 @@ - (void)setUpDateFormatter
 
 - (void)logMessage:(DDLogMessage *)logMessage
 {
-	logMessage = [self formatLogMessage:logMessage];
-	[self scheduleLogMessage:logMessage];
-}
-
-- (DDLogMessage *)formatLogMessage:(DDLogMessage *)logMessage
-{
-	if (formatter)
-	{
-		NSString *logMsg = [formatter formatLogMessage:logMessage];
-		if (logMsg != nil)
-		{
-			logMessage->logMsg = logMsg;
-		}
-		else
-		{
-			logMessage = nil;
-		}
-	}
-	return logMessage;
+    NSString *logMsg = [formatter formatLogMessage:logMessage];
+    
+    if (logMsg)
+    {
+        logMessage->logMsg = logMsg;
```

Comment:
This looks better, however `logMessage` is now an output argument. What if you copy `logMessage` and modify the copy instead of the argument? If you can not use `[logMessage copy]` (it was implemented for the first time for the version [1.6.5](https://github.com/CocoaLumberjack/CocoaLumberjack/releases/tag/1.6.5)), would it be an option to implement your own `copy` in a category?